lib/api/objectPutRetention.js

```diff
@@ -26,6 +27,12 @@ const REPLICATION_ACTION = 'PUT_RETENTION';
 function objectPutRetention(authInfo, request, log, callback) {
     log.debug('processing request', { method: 'objectPutRetention' });
 
+    const err = validateChecksumsNoChunking(request.headers, request.post);
+    if (err && err.error !== ChecksumError.MissingChecksum) {
+        log.debug(err, { error: errors.BadDigest });
```

Comment:
I'm not sure we need to log the error that we return, better to have a string like this?
```suggestion
        log.debug('failed checksum validation', err);
```

Same for all other places

Reply:
ok

Reply:
done